src/utilities/hooks/ModalPaymentContext/reducers.ts

```diff
@@ -47,8 +50,28 @@ const ModalPaymentReducer = (state: ModalPaymentState, action: Action) => {
         amount: defaultState.amount,
         purchaseType: null,
       };
+    case SET_LUC_DATA:
+      return {
+        ...state,
+        lucData: {
+          ...state.lucData,
+          [payload.key]: payload.value,
+        },
+      };
     case CLEAR_FORMS:
       return defaultState;
+    case CLEAR_ADDRESS:
+      return {
+        ...state,
+        lucData: {
+          ...state.lucData,
+          fullName: '',
```

Comment:
do you need to also clear `firstName`, etc.?

Reply:
firstName is used for the engraving of the lantern, fullname is used for the billing address